pkg/providers/vsphere/vsphere.go

```diff
@@ -443,6 +450,80 @@ func (p *vsphereProvider) validateMachineConfigsNameUniqueness(ctx context.Conte
 	return nil
 }
 
+func (p *vsphereProvider) getCurrentMachineConfigDatastoreUsage(ctx context.Context, machineConfig *v1alpha1.VSphereMachineConfig, vsphereClusterSpec *Spec, cluster *types.Cluster) (diskGiB float64, err error) {
+	if cluster == nil {
+		return 0, nil
+	}
+	em, err := p.providerKubectlClient.SearchVsphereMachineConfig(ctx, machineConfig.Name, cluster.KubeconfigFile, vsphereClusterSpec.Cluster.GetNamespace())
+	if err != nil {
+		return 0, err
+	}
+	if len(em) > 0 {
+		return float64(em[0].Spec.DiskGiB), nil
+	}
+	return 0, nil
+}
+
+func (p *vsphereProvider) calculateDatastoreUsage(ctx context.Context, machineConfig *v1alpha1.VSphereMachineConfig, vsphereClusterSpec *Spec, cluster *types.Cluster, usage map[string]*datastoreUsage) error {
+	availableSpace, needGiB, err := p.getMachineConfigDatastoreRequirements(ctx, machineConfig, vsphereClusterSpec.Cluster.Spec.ControlPlaneConfiguration.Count)
```

Comment:
This function is being called for cp, workers and etcd separately right? We shouldn't hardcode the control plane node count (`vsphereClusterSpec.Cluster.Spec.ControlPlaneConfiguration.Count`) here